apps/nextjs/src/stores/chatStore/index.ts

```diff
@@ -12,56 +15,53 @@ type ChatStore = {
   isLoading: boolean;
   stableMessages: ParsedMessage[];
   streamingMessage: ParsedMessage | null;
+  queuedUserAction: string | null;
+  isExecutingAction: boolean;
 
   // Actions
+  append: (
+    message: Message | CreateMessage,
+    chatRequestOptions?: ChatRequestOptions | undefined,
+  ) => Promise<string | null | undefined>;
+  reload: () => void;
+  setAppend: (
+    append: (
+      message: Message | CreateMessage,
+      chatRequestOptions?: ChatRequestOptions | undefined,
+    ) => Promise<string | null | undefined>,
+  ) => void;
+  setReload: (reload: () => void) => void;
+  stop: () => void;
   setMessages: (messages: AiMessage[], isLoading: boolean) => void;
   setIsLoading: (isLoading: boolean) => void;
+  setStop: (stop: () => void) => void;
+  queueUserAction: (action: string) => void;
+  executeQueuedAction: () => Promise<void>;
 };
 
 export const useChatStore = create<ChatStore>((set, get) => ({
   // From AI SDK
   isLoading: false,
   stableMessages: [],
   streamingMessage: null,
+  queuedUserAction: null,
+  isExecutingAction: false,
 
   // Actions
-  setMessages: (messages, isLoading) => {
-    if (!isLoading) {
-      // All messages are stable
-      const nextStableMessages = getNextStableMessages(
-        messages,
-        get().stableMessages,
-      );
-      set({
-        ...(nextStableMessages && {
-          stableMessages: nextStableMessages,
-        }),
-        streamingMessage: null,
-      });
-    } else {
-      // The latest message is streaming, previous messages are stable
-      const currentMessageData = messages[messages.length - 1];
-      invariant(currentMessageData, "Should have at least one message");
-      const streamingMessage = parseStreamingMessage(
-        currentMessageData,
-        get().streamingMessage,
-      );
-
-      const stableMessageData = messages.slice(0, messages.length - 1);
-      const nextStableMessages = getNextStableMessages(
-        stableMessageData,
-        get().stableMessages,
-      );
-
-      set({
-        streamingMessage,
-        ...(nextStableMessages && {
-          stableMessages: nextStableMessages,
-        }),
-      });
-    }
+  stop: () => {},
+  append: async () => "",
+  reload: () => {},
+  setAppend: (append) => set({ append }),
+  setReload: (reload) => set({ reload }),
```

Comment:
I think we should group these functions used for config elsewhere, as they aren't really actions that we should think about when it comes to use the store